### PR TITLE
Don't skip reinstall if Rose CLI opts set

### PR DESCRIPTION
**Pull request by wxtim**
This bug is not documented elsewhere.

Previous behavior was to ignore these if there is not file change.

However this does not work as expected in the case where a user changes rose command line arguments, or wishes to clear CLI args from a previous install.

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Tests have been updated
- [x] `CHANGES.md` entry included if this is a change that can affect users

### CODE REVIEW

File: cylc/flow/scripts/reinstall.py [hunk 1]
```diff
@@ -170,12 +171,16 @@ def reinstall_cli(
...
+            if (
+                not reinstall(
+                    opts,
+                    workflow_id,
+                    source,
+                    run_dir,
+                    dry_run=True,
+                )
+                and not has_rose_cli_opts(opts)
+                and not opts.clear_rose_install_opts
```
[the remark this answers is not in the document]
Comment (wxtim) [reply cylc/flow/scripts/reinstall.py:183]: Why not.

File: cylc/flow/option_parsers.py [hunk 2]
```diff
@@ -583,3 +583,20 @@ def __call__(self, **kwargs) -> Values:
...
+        hasattr(opts, rose_opt) and getattr(opts, rose_opt)
+        for rose_opt in [
+            'opt_conf_keys',
+            'defines',
+            'rose_template_vars',
+            'clear_rose_install_opts'
+        ]
+    ):
+        return True
+    return False
```
Comment (oliver-sanders) [inline cylc/flow/option_parsers.py:602]:
```suggestion
    return any(
        hasattr(opts, rose_opt) and getattr(opts, rose_opt)
        for rose_opt in [
            'opt_conf_keys',
            'defines',
            'rose_template_vars',
            'clear_rose_install_opts'
        ]
    )
```

**Review by oliver-sanders — COMMENTED** [review]
I've just realised, after your recent reinstall change (which effectively marks `rose-suite.conf` as changed no matter what) is this still needed?

I think any workflow with a `rose-suite.conf` will now be considered for reinstallation irrespective of whether anything has been changed or not?

**Comment by wxtim** [comment]
> I've just realised, after your recent reinstall change (which effectively marks `rose-suite.conf` as changed no matter what) is this still needed?
> 
> I think any workflow with a `rose-suite.conf` will now be considered for reinstallation irrespective of whether anything has been changed or not?

Yes

**Pull request closed without merging**